ibazel/ibazel.go

```diff
@@ -300,38 +353,112 @@ func (i *IBazel) iteration(command string, commandToRun runnableCommand, targets
 	case RUN:
 		fmt.Fprintf(os.Stderr, "%sing %s\n", strings.Title(command), joinedTargets)
 		i.beforeCommand(targets, command)
-		err := commandToRun(targets...)
-		i.afterCommand(targets, command, err == nil)
+		outputBuffer, err := commandToRun(targets...)
+		i.afterCommand(targets, command, err == nil, outputBuffer)
+		i.state = WAIT
+	}
+}
+
+func (i *IBazel) iterationMultiple(command string, commandToRun runnableCommands, targets []string, debugArgs [][]string, argsLength int) {
```

Comment:
wow a ton of code is added to this file - is it possible to simplify?

Reply:
It's possible. Since `test`, `build`, and `run` share a few functions, `mrun` has its own functions with different signatures but do similar things, we can try to merge those functions and make `mrun` a special case. It requires a lot of works and involves changing code for `test` and `build`. Should we do it?